templates/grub.cfg.erb

```diff
@@ -0,0 +1,8 @@
+insmod regexp
+
+set net_pxe_mac=$net_default_mac
+regexp --set=1:m1 --set=2:m2 --set=3:m3 --set=4:m4 --set=5:m5 --set=6:m6 '^([0-9a-f]{1,2})\:([0-9a-f]{1,2})\:([0-9a-f]{1,2})\:([0-9a-f]{1,2})\:([0-9a-f]{1,2})\:([0-9a-f]{1,2})' "$net_default_mac"
+mac_dashed=${m1}-${m2}-${m3}-${m4}-${m5}-${m6}
+
+mac_configfile=$prefix/grub.cfg-01-$mac_dashed
+source "$mac_configfile"
```

Comment:
Just wondering, a user reports that grub2 from Ubuntu 16.04 does not work with this, `$net_default_mac` is unset for him. Any idea? Maybe to `insmod efinet` or somethingg? It looks like this variable did not carry over from Grub to Grub2 according official docs.